addons/mail/static/src/discuss/call/common/call_menu.js

```diff
@@ -9,6 +9,38 @@ export class CallMenu extends Component {
     setup() {
         super.setup();
         this.rtc = useState(useService("discuss.rtc"));
+        this.state = useState({
+            systrayIcon: undefined,
+        });
+        this.lastActions = [];
+        this.actionsIcons = [
+            "fa-desktop",
+            "fa-video-camera",
+            "fa-hand-paper-o",
+            "fa-deaf",
+            "fa-microphone-slash",
+        ];
+        useEffect(
+            (...actions) => {
+                for (let i = 0; i < actions.length; i++) {
+                    if (actions[i] !== this.lastActions[i]) {
+                        this.state.systrayIcon = actions[i]
+                            ? this.actionsIcons[i]
+                            : this.actionsIcons[actions.findIndex((v) => v)];
+                        break;
+                    }
+                }
+
+                this.lastActions = actions;
+            },
+            () => [
+                this.rtc.state.sendScreen,
+                this.rtc.state.sendCamera,
+                this.rtc.selfSession?.raisingHand,
+                this.rtc.selfSession?.isDeaf,
+                this.rtc.selfSession?.isSelfMuted,
+            ]
+        );
```

Comment:
This doesn't match the specs: here it keeps the icon of the last observed call action.

In the specs, it states that the icon should display the state of the call. There are some similarities, but also differences:

- If I share screen, then mute, then unmute, I would expect to show screen-sharing icon, not mic icon.
- if I raise my hand and then un-raise, I would expect to show the current state (e.g. mic icon), not the hand icon.

So each action, whether enabled or disabled, should be put in a stack. When an action is off, it should be dequeued from the state.

---------------

On another node, in terms of code maintenance, it'd best to improve Locality of Concerns by defining call actions with their icons. Until now call icons were only relevant in a single template, but you now need to reuse.
So this incentives to define call actions like thread and message actions.

Reply:
> * If I share screen, then mute, then unmute, I would expect to show screen-sharing icon, not mic icon.
> * if I raise my hand and then un-raise, I would expect to show the current state (e.g. mic icon), not the hand icon.

This is the behaviour I observe with my commit.